autoload/codefmt.vim

```diff
@@ -279,9 +282,23 @@ function! s:GetFormatter(...) abort
     if !empty(l:default_formatters)
       let l:formatter = l:default_formatters[0]
     else
-      call maktaba#error#Shout(
-          \ 'Not available. codefmt doesn''t have a default formatter for ' .
-          \ 'this buffer.')
+      " Check if we have formatters that are not avialable for some reason.
+      " Report a better error message in that case.
+      let l:unavailable_formatters = filter(
+            \ copy(l:formatters), 'v:val.AppliesToBuffer()')
+      let l:error = ''
```

Comment:
You can replace this whole thing with `let l:error = join(map(copy(l:unavailable_formatters), 's:GetSetupInstructions(v:val)'), "\n")`. Doesn't have to be all one line, but using `join()` would be a big improvement, and `map()` probably would too.